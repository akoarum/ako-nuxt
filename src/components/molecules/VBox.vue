<template>
  <div class="vBox">
    <VHeading :level="level" :visual="3" class="vBox__title">
      <slot name="title" />
    </VHeading>
    <div class="vBox__content">
      <slot />
    </div>
  </div>
</template>

<script>
import { VHeading } from '~/components/atoms'
export default {
  components: { VHeading },
  props: {
    level: { type: Number, default: 2 }
  }
}
</script>

<style lang="scss" scoped>
.vBox {
  border: 1px solid $color-border;

  @include media-min {
    padding: 10px 16px 14px;
  }

  @include media-max {
    padding: 8px 14px 12px;
  }
}

.vBox__title {
  border-bottom: 1px solid $color-border;

  @include media-min {
    padding: 0 2px 6px;
  }

  @include media-max {
    padding: 0 2px 3px;
  }
}

.vBox__content {
  padding: 8px 2px 0;
}
</style>
