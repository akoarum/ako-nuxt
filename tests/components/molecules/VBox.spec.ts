import { shallowMount } from '@vue/test-utils'
import VBox from '~/components/molecules/VBox.vue'

describe('VBox', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(VBox, {
      propsData: {
        title: 'テスト'
      }
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })
    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
