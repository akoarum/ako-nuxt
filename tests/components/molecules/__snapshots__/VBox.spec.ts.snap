// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VBox initialized snapshot 1`] = `
<div
  class="vBox"
>
  <vheading-stub
    class="vBox__title"
    level="2"
    visual="3"
  >
    
    テスト
  
  </vheading-stub>
   
  <div
    class="vBox__content"
  />
</div>
`;
